Typescript types for `setValidatorCompiler` mismatch

## 🐛 Bug Report

Typescript typings no longer work for setting ajv as a schema validator (even when using the code sample from the README).

## To Reproduce

Steps to reproduce the behavior:

```typescript
  const ajv = new Ajv({
    // the fastify defaults (if needed)
    removeAdditional: false,
    useDefaults: true, // replace missing properties and items with the values from corresponding default keyword
    coerceTypes: true, // change data type of data to match type keyword
    allErrors: true, // check for all errors
    nullable: true,
  })
  fastify.setValidatorCompiler(({ schema }) => {
    return ajv.compile(schema) 
  })
```

This yields: 

```
src/setupFastify.ts:74:47 - error TS2322: Type 'ValidateFunction' is not assignable to type 'FastifyValidationResult'.
  Types of property 'errors' are incompatible.
    Type 'ErrorObject[]' is not assignable to type 'FastifySchemaValidationError[]'.
      Type 'ErrorObject' is not assignable to type 'FastifySchemaValidationError'.
        Property 'message' is optional in type 'ErrorObject' but required in type 'FastifySchemaValidationError'.

74   fastify.setValidatorCompiler(({ schema }) => ajv.compile(schema))
                                                 ~~~~~~~~~~~~~~~~~~~

  node_modules/fastify/types/schema.d.ts:35:37
    35 export type FastifySchemaCompiler = (routeSchema: FastifyRouteSchemaDef) => FastifyValidationResult
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    The expected type comes from the return type of this signature.
```

It would be nice to get an up to date sample on how to configure fastify with different ajv settings.

## Expected behavior

Typescript compilation succeeds 

## Your Environment

- *node version*: 12.13
- *fastify version*: 3.4.1
- *typescript version*: 4.0.3
- *os*: Mac
- *any other relevant information*

--- a/test/types/schema.test-d.ts
+++ b/test/types/schema.test-d.ts
@@ -1,5 +1,6 @@
 import { expectAssignable, expectError } from 'tsd'
 import fastify, { FastifyInstance } from '../../fastify'
+import Ajv = require('ajv')
 
 const server = fastify()
 
@@ -30,6 +31,10 @@ expectAssignable<FastifyInstance>(server.get(
   {},
   () => { }
 ))
+
+expectAssignable<FastifyInstance>(server.setValidatorCompiler(({ schema }) => {
+  return new Ajv().compile(schema)
+}))
 
 expectError(server.get(
   '/unknown-schema-prop',

--- a/types/schema.d.ts
+++ b/types/schema.d.ts
@@ -21,11 +21,11 @@ export interface FastifyRouteSchemaDef {
 }
 
 export interface FastifyValidationResult {
-  errors?: FastifySchemaValidationError[];
+  errors?: FastifySchemaValidationError[] | null;
 }
 
 export interface FastifySchemaValidationError {
-  message: string;
+  message?: string;
   dataPath: string;
 }
 

